From d0be0df98679b7a9a30ba74c065ed30301e2bd22 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 15 Jan 2020 13:59:57 +0000
Subject: [PATCH] driver: char: rpivid: Destroy the legacy device on
 remove

The legacy name support created a new device that was never destroyed.
If the driver was unloaded and reloaded, it failed due to the
device already existing.

Fixes: "75f1d14 driver: char: rpivid - also support legacy name"
Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/char/broadcom/rpivid-mem.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/char/broadcom/rpivid-mem.c
+++ b/drivers/char/broadcom/rpivid-mem.c
@@ -233,6 +233,7 @@ static int rpivid_mem_remove(struct plat
 	struct device *dev = &pdev->dev;
 	struct rpivid_mem_priv *priv = platform_get_drvdata(pdev);
 
+	device_destroy(priv->class, priv->devid + 1);
 	device_destroy(priv->class, priv->devid);
 	class_destroy(priv->class);
 	cdev_del(&priv->rpivid_mem_cdev);
